lib/uicc_lwm2m/asn1_decode.c

```diff
@@ -0,0 +1,83 @@
+/*
+ * Copyright (c) 2024 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
+ */
+
+#include <zephyr/sys/util.h>
+
+#include "asn1_decode.h"
+
+bool asn1_dec_head(asn1_ctx_t *ctx, uint8_t *tag, size_t *len)
+{
+	uint32_t hlen = 2; /* Minimum two bytes for header */
+
+	if (ctx->error || ((ctx->offset + hlen) > ctx->length)) {
+		/* Error detected or out of data (happens at end of sequence) */
+		return false;
+	}
+
+	*tag = ctx->asnbuf[ctx->offset++];
+	*len = ctx->asnbuf[ctx->offset++];
+
+	if ((*tag & 0x1F) == 0x1F) {
```

Comment:
replace magic numbers with defines

Reply:
This is code parsing ASN.1 tag and len. I think it will make the code more unreadable if changing bit checks with defines.